### DISCUSSION TITLE
Should this package be retired in favor of pytest-httpserver?

### DISCUSSION THREAD

**Discussion by diazona** [pytest-dev/pytest-localserver#46] [General]
I wanted to move the discussion here from #43: the idea was put forth that since [pytest-httpserver](https://github.com/csernazs/pytest-httpserver) provides most of the same functionality and is generally more mature than this project, would it make sense to stop developing this package and transition people to using pytest-httpserver instead? Let's continue the discussion here.

**Comment by csernazs** [comment]
Hi there,

I think I would need a little bit of clarification.

What do you mean by "transition people to using pytest-httpserver"? Do you maintain a code which uses localserver and needs to be ported to httpserver?

How does it compares having a project hosted under "pytest-dev" and elsewhere? What are the requirements for a project be hosted under "pytest-dev"?

Both pytest-localserver and pytest-httpserver are available on pypi and I think people can decide which one they want to use.

**Reply by diazona** [reply]
The idea that was proposed in #43, as I understand it, was to stop developing pytest-localserver (i.e. archive the project and mark it as deprecated/obsolete in PyPI), and recommend that anyone who wants an HTTP server fixture to use in pytest should use pytest-httpserver instead. That's what I mean by "transition people to using pytest-httpserver".

That's contingent on the assumption that, given the choice between pytest-localserver and pytest-httpserver, most people would choose to use pytest-httpserver. Of course, there is some disagreement about whether that's true.

Note that moving pytest-httpserver under the pytest-dev organization is not necessarily a prerequisite for this to happen. You can make your own decision about whether you'd be interested in doing that, and that decision would be one factor we take into account when we decide whether to continue developing pytest-localserver.

**Reply by diazona** [reply]
> How does it compares having a project hosted under "pytest-dev" and elsewhere? What are the requirements for a project be hosted under "pytest-dev"?

Based on my experience, the requirements for hosting a project under pytest-dev are pretty minimal. Of course, it needs to be a project that the pytest-dev organization is interested in hosting, but aside from that, all we really needed to do was make sure that there are multiple people with maintainer access to the project, both on Github and on PyPI, including some pytest developers. The pytest developers usually do not get very involved in development of the package, but they need to have access so that they can find new maintainers if the rest of us drop off the grid.

I'd suggest talking to someone like @nicoddemus if you want more information about how this would work.

**Reply by The-Compiler** [reply]
With my pytest maintainer hat on: I believe especially in a situation like this, something being the designated replacement of something already in pytest-dev, this would be pretty much a no-brainer.

The whole process is documented here: [Submitting Plugins to pytest-dev](https://docs.pytest.org/en/latest/contributing.html#submitting-plugins-to-pytest-dev).

**Comment by The-Compiler** [comment]
As mentioned in https://github.com/pytest-dev/pytest-localserver/issues/43#issuecomment-1186126887, `pytest-localserver` can also host an SMTP server, for which (as far as I'm aware) no alternative is available. Unless we really don't find anyone who wants to maintain it, I feel like we should have an alternative plugin in place for that part as well - as it stands, pytest-httpserver is *not* a full replacement for what pytest-localserver does.

**Reply by RonnyPfannschmidt** [reply]
i stumbled uppon https://github.com/bebleo/smtpdfix

**Reply by diazona** [reply]
Makes sense. Personally (if this "retirement" happens) I wouldn't mind splitting the SMTP server functionality off into its own separate package either, maybe pytest-smtpserver, but given that there is an existing package that does the same thing (and which appears to be maintained), the considerations for that are probably about the same as with the http part.
